app/controllers/admin/events_controller.rb

```diff
@@ -23,11 +23,21 @@ def index
       @tracks_distribution_confirmed = @conference.tracks_distribution(:confirmed)
       @event_distribution = @conference.event_distribution
       @scheduled_event_distribution = @conference.scheduled_event_distribution
+      @file_name = "events_for_#{@conference.short_title}"
+      @my_var = params[:my_var]
 
       respond_to do |format|
         format.html
         # Explicity call #to_json to avoid the use of EventSerializer
         format.json { render json: Event.where(state: :confirmed, program: @program).to_json }
+        format.xlsx do
+          render_xlsx_file
```

Comment:
Please try some more googling about this. I am sure you can figure out the solution :-)